app/api/client.ts

```diff
@@ -49,6 +53,24 @@ export const useApiMutation = getUseApiMutation(api.methods)
 export const usePrefetchedQuery = <TData>(options: UseQueryOptions<TData, ApiError>) =>
   ensurePrefetched(useQuery(options), options.queryKey)
 
+/**
+ * Extends React Query's `QueryClient` with a couple of API-specific methods.
+ * Existing methods are never modified.
+ */
+class QueryClient extends QueryClientOrig {
+  /**
+   * Invalidate all cached queries for a given endpoint.
+   *
+   * Note that we only take a single argument, `method`, rather than allowing
+   * the full query key `[query, params]` to be specified. This is to avoid
+   * accidentally overspecifying and therefore failing to match the desired query.
+   * The params argument can be added in if we ever have a use case for it.
+   */
+  invalidateEndpoint(method: keyof typeof api.methods) {
+    this.invalidateQueries({ queryKey: [method] })
+  }
+}
+
```

Comment:
This approach feels good, much better than a separate `invalidate` function (which I tried in 622bc80c076211d3d1adc10e18a7fa4dc6fa0bc5) or a separate `apiQueryClient`, like we have now. I like that it adds a new method and does not touch existing methods.